ntcore/src/main/native/cpp/InstanceImpl.h

```diff
@@ -77,6 +81,8 @@ class InstanceImpl {
   std::shared_ptr<NetworkServer> m_networkServer;
   std::shared_ptr<INetworkClient> m_networkClient;
   std::vector<std::pair<std::string, unsigned int>> m_servers;
+  std::optional<int64_t> m_serverTimeOffset{0};
```

Comment:
Do we want to default this to zero or empty?

Reply:
Zero, because unless you're operating as a client, 0 is the correct value.

Reply:
So when is it empty?
Also, does an optional containing `0` resolve to `true` when converted to bool?

Reply:
It’s empty when you’re a client and you’re not connected or not yet gotten a time ping response from the server.

Yes an optional containing 0 is true.

Reply:
Ended up changing the contract so it's empty unless it's running in server mode (in which case it's 0) or in client mode when it's connected to a server.